packages/observe-sequence/observe_sequence_tests.js

```diff
@@ -93,6 +93,37 @@ runOneObserveSequenceTestCase = function (test, sequenceFunc,
              compress(EJSON.stringify(expectedCallbacks, {canonical: true, indent: true})));
 };
 
+// This creates an object subclassed from Array
+const hasProp = {}.hasOwnProperty;
+const extend = function(child, parent) {
+  function ctor() {
+    this.constructor = child;
+  }
+  for(let key in parent) {
+    if(hasProp.call(parent, key)) {
+      child[key] = parent[key];
+      console.log('key: ' + key);
+    }
+  }
+  ctor.prototype = parent.prototype;
+  child.prototype = new ctor();
+  child.__super__ = parent.prototype;
+  return child;
+};
+const ArraySubclass = (function (superClass) {
+  extend(ArraySubclass, superClass);
+
+  function ArraySubclass() {
+    if (arguments.length > 0) {
+      const items = this.slice.call(arguments, 0);
+      this.splice.apply(this, [0, 0].concat(this.slice.call(items)));
+    }
+  }
+
+  return ArraySubclass;
+
+})(Array);
+
```

Comment:
I realize this sub-classing of a built-in is difficult, and it'll be ideal to remove eventually.  I'm not sure if it would be sufficient for this test, but the following might be more succinct and work?

```js
function ArraySubclass() {
  const Fp = Function.prototype;
  const inst = new (Fp.bind.apply(Array, [null].concat(Array.from(arguments))));
  inst.__proto__ = ArraySubclass.prototype;
  return inst;
}
ArraySubclass.prototype = Object.create(Array.prototype);
```

and then I believe the code below would be fine:

```js
var seq = new ArraySubclass(1, 1, 2);
```